Project/Client/LogUI.cpp

```diff
@@ -0,0 +1,311 @@
+    }
+
+    for (int i = 0; i < 3; ++i)
+    {
+        if (m_LogTimePrint[i + 1])
+            m_iLogTimeMask |= (1 << i);
+        else
+            m_iLogTimeMask &= ~(1 << i);
+    }
+
+    CLogMgr::GetInst()->SetTimeMask(m_iLogTimeMask);
+}
+
+bool LogUI::CheckSearchDisplay(string msg)
+{    
+    return m_TextFilter.PassFilter(msg.c_str());
+}
+
+
+bool LogUI::CheckLogLvDispaly(Log_Level Loglv)
+{
+    return (UINT)Loglv & m_iLoglvMask;
+}
+
+void LogUI::tick()
+{
+}
+
+void LogUI::render_update()
+{
+    m_vectLog = CLogMgr::GetInst()->GetLogs();
+
+    // 복사 버튼
+    ImGui::PushStyleColor(ImGuiCol_Button, ImVec4(0.3f, 0.3f, 0.3f, 1.0f)); // 기본 배경색
+    ImGui::PushStyleColor(ImGuiCol_ButtonHovered, ImVec4(1.f, 0.f, 0.f, 1.0f)); // 호버 배경색도 기본 배경색과 동일하게 설정
+    ImGui::PushStyleColor(ImGuiCol_ButtonActive, ImVec4(1.f, 0.f, 0.f, 1.0f));
+    if (ImGui::Button("CopyLog"))
+    {
+        CLogMgr::GetInst()->CopyLog();
+    }
+    ImGui::PopStyleColor(3);
+
+    ImGui::SameLine();
+
+
+    // 삭제 버튼
+    ImGui::PushStyleColor(ImGuiCol_Button, ImVec4(0.3f, 0.3f, 0.3f, 1.0f)); // 기본 배경색
+    ImGui::PushStyleColor(ImGuiCol_ButtonHovered, ImVec4(1.f, 0.f, 0.f, 1.0f)); // 호버 배경색도 기본 배경색과 동일하게 설정
+    ImGui::PushStyleColor(ImGuiCol_ButtonActive, ImVec4(1.f, 0.f, 0.f, 1.0f));
+    if (ImGui::Button("DeleteLog"))
+    {
+        CLogMgr::GetInst()->ClearLog();
+    }
+    ImGui::PopStyleColor(3);
+
+    // 필터 체크박스
+    
+    if (ImGui::Checkbox("All", &m_LoglvCheckBox[0]))
+    {
+        if (m_LoglvCheckBox[0])
+        {
+            m_LoglvCheckBox[1] = false;
+            m_LoglvCheckBox[2] = false;
+            m_LoglvCheckBox[3] = false;
+        }
+    }
+    ImGui::SameLine();
+
+    if (ImGui::Checkbox("Info", &m_LoglvCheckBox[1]))
+    {
+        if (m_LoglvCheckBox[0])
+            m_LoglvCheckBox[0] = false;
+    }
+    ImGui::SameLine();
+
+    if (ImGui::Checkbox("Warn", &m_LoglvCheckBox[2]))
+    {
+        if (m_LoglvCheckBox[0])
+            m_LoglvCheckBox[0] = false;
+    }
+
+    ImGui::SameLine();
+
+    if (ImGui::Checkbox("Error", &m_LoglvCheckBox[3]))
+    {
+        if (m_LoglvCheckBox[0])
+            m_LoglvCheckBox[0] = false;
+    }
+
+    CheckLevelMask();
+
+
+    // 시간 관련 표기
+    if (ImGui::Checkbox("No Time", &m_LogTimePrint[0]))
+    {
+        if (m_LogTimePrint[0])
+        {
+            for (int i = 1; i < 4; ++i)
+            {
+                m_LogTimePrint[i] = false;
+            }
+        }
+    }
+    ImGui::SameLine();
+    
+    if (ImGui::Checkbox("Sec", &m_LogTimePrint[1]))
+    {
+        if (m_LogTimePrint[0])
+            m_LogTimePrint[0] = false;
+    }
+    ImGui::SameLine();
+
+    if (ImGui::Checkbox("Min", &m_LogTimePrint[2]))
+    {
+        if (m_LogTimePrint[0])
+            m_LogTimePrint[0] = false;
+    }
+    ImGui::SameLine();
+
+    if (ImGui::Checkbox("Hour", &m_LogTimePrint[3]))
+    {
+        if (m_LogTimePrint[0])
+            m_LogTimePrint[0] = false;
+    }
```

Comment:
이 또한...
내 설명을 이해 못하고 작성한건가?
```suggestion
        if (m_LogTimePrint[1])
            m_LogTimePrint[0] = false;
    }
    ImGui::SameLine();

    if (ImGui::Checkbox("Min", &m_LogTimePrint[2]))
    {
        if (m_LogTimePrint[2])
            m_LogTimePrint[0] = false;
    }
    ImGui::SameLine();

    if (ImGui::Checkbox("Hour", &m_LogTimePrint[3]))
    {
        if (m_LogTimePrint[3])
            m_LogTimePrint[0] = false;
    }
```